File: code/ch19/19.1.4. 189.rotate.cpp
#include <vector>
using namespace std;

class Solution {
public:
  vector<int> rotate(vector<int>& nums, int k) {
    int n = nums.size(), offset = k % n;
    for (int i = 0; i < n; i++) nums.push_back(nums[i]);
    vector<int> ans;
    for (int i = n - offset; i < n * 2 - offset; i++)
      ans.push_back(nums[i]);
    return ans;
  }
};

File: code/ch19/19.1.4. 189.rotate.java
public class Solution {
  public int[] rotate(int[] nums, int k) {
    int n = nums.length, offset = k % n;
    int[] ano_nums = new int[2 * n];
    for (int i = 0; i < n; i++)
      ano_nums[i] = ano_nums[n + i] = nums[i];
    int[] ans = new int[n];
    for (int i = n - offset; i < n * 2 - offset; i++)
      ans[i - n + offset] = ano_nums[i];
    return ans;
  }
}
